enhancements/network/ip-interface-selection.yaml

```diff
@@ -0,0 +1,278 @@
+services would not have to figure out the correct interface to use for a given
+IP. However, this is problematic for use cases like OVNKubernetes because on
+initial boot the IP will be on the interface itself, while on subsequent boots
+the address will be on br-ex. Because the IP may move after
+nodeip-configuration runs, we don't want to persist the interface.
+
+### Risks and Mitigations
+
+- There is some risk to changing the order of critical system services like
+  nodeip-configuration and configure-ovs. This will not affect deployments
+  that do not use OVNKubernetes as their CNI, but since we intend that to be
+  the default going forward it is a significant concern.
+
+  We intend to mitigate this risk by first merging the order change without
+  any additional changes included in this design. This way, if any races
+  between services are found once the change is being tested more broadly
+  it will be easy to revert.
+
+  We will also test the ordering change as much as possible before merging
+  it, but it's unlikely we can exercise it to the same degree that running
+  across hundreds of CI jobs per day will.
+
+- Currently all host services are expected to listen on the same IP and
+  interface. If at some point in the future we need host services listening
+  on multiple different interfaces, this may not work. However, because we
+  are centralizing all IP selection logic in nodeip-configuration, it should
+  be possible to extend that to handle multiple interfaces if necessary.
+
+### Drawbacks
+
+This design only considers host networking, and it's likely that in the future
+we will want a broader feature that provides an interface to configure pod
+traffic routing as well. However, if/when such a feature is implemented it
+should be able to use the same configuration interface for host services
+that deployers would use directly after this is implemented.
+
+Additionally, there are already ways to [implement traffic steering for pod
+networking.](https://youtu.be/EpbUWwjadYM) We may at some point want to
+integrate them more closely, but host networking is currently a much bigger
+pain point and worth addressing on its own.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Currently this only applies to UPI clusters deployed using platform None.
+  Do we need something similar for IPI?
+  - Answer: Yes. In IPI deployments the VIP is always used for node ip
+    selection. In an environment where the VIP subnet is isolated this could
+    result in the node IP and the interface selected by configure-ovs not
+    matching.
+
+- In UPI deployments it is also possible to set the Node IP by manually writing
+  configuration files for Kubelet and CRIO. Trying to look for all possible
+  ways a user may have configured those services seems complex and error-prone.
+  Can we just require them to use this mechanism if they want custom IP
+  selection?
```

Comment:
we're waiting for an answer to this question, right ? 

Personally, it makes sense to me, but I see this as part of unifying IP and/or interface selection.

Reply:
I guess we could proceed without making this statement, but I think we'll continue to have problems when people use other mechanisms to set the node ip.